src/tools/auth0/handlers/emailProvider.ts

```diff
@@ -30,34 +31,38 @@ export default class EmailProviderHandler extends DefaultHandler {
   async processChanges(assets: Assets): Promise<void> {
     const { emailProvider } = assets;
 
-    // Do nothing if not set
     if (!emailProvider) return;
 
-    if (Object.keys(emailProvider).length > 0) {
-      let existing = await this.getType();
-
-      // Check for existing Email Provider
-      if (existing.name) {
-        if (existing.name !== emailProvider.name) {
-          // Delete the current provider as it's different
-          await this.client.emailProvider.delete();
-          this.didDelete(existing);
-          existing = {};
-        }
+    let existing = await this.getType();
+
+    if (Object.keys(emailProvider).length === 0) {
+      if (this.config('AUTH0_ALLOW_DELETE') === true) {
+        logger.warn(
+          'Setting email provider to empty object will delete the provider in future versions. The current inability to delete is considered a bug. For more details see: https://github.com/auth0/auth0-deploy-cli/issues/653'
+        );
       }
+      return;
```

Comment:
This early return replaces the nested if statement that existed previously, thus causing a large diff.